modules/allowances/test/test-helpers/deploySafeProxy.ts

```diff
@@ -1,27 +1,10 @@
-import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers'
-import { AbiCoder, concat, getCreate2Address, Interface, keccak256, ZeroAddress, ZeroHash } from 'ethers'
+import { Interface, ZeroAddress } from 'ethers'
 
-import { ArtifactSafe, ArtifactSafeProxy, ArtifactSafeProxyFactory } from './artifacts'
+import ArtifactSafe from '../../build/artifacts/@safe-global/safe-contracts/contracts/Safe.sol/Safe.json'
+import ArtifactSafeZk from '../../build/artifacts-zk/@safe-global/safe-contracts/contracts/Safe.sol/Safe.json'
 
-export default async function deploySafeProxy(
-  factory: string,
-  mastercopy: string,
-  owner: string,
-  deployer: SignerWithAddress,
-): Promise<string> {
-  const initializer = calculateInitializer(owner)
-
-  const iface = new Interface(ArtifactSafeProxyFactory.abi)
-  await deployer.sendTransaction({
-    to: factory,
-    data: iface.encodeFunctionData('createProxyWithNonce', [mastercopy, initializer, ZeroHash]),
-  })
-
-  return calculateProxyAddress(initializer, factory, mastercopy)
-}
-
-function calculateInitializer(owner: string): string {
-  const iface = new Interface(ArtifactSafe.abi)
+export function calculateInitializer(owner: string, zkSync: boolean = false): string {
+  const iface = new Interface(zkSync ? ArtifactSafeZk.abi : ArtifactSafe.abi)
```

Comment:
The abi should be the same regardless of the network, why is it needed?

Reply:
correct.